THUB-17 | @jdwjdwjdw | Update wysiwyg styles, hero fixup, search results page button fixup

jdwjdwjdw opened the pull request:
# READY FOR REVIEW

# Summary
- [THUB-17](https://stanfordits.atlassian.net/browse/THUB-17): Text Area
  - See [Figma](https://www.figma.com/design/oFS2wdu1DqU08czjesLPa6/Taiwan-Science-and-Technology-Hub?node-id=4627-3052&m=dev)
  - There are some headings not that will need this badge that aren't in the wysiwyg - I'll address those in a separate PR. Design also asked for the badge to not be applied to headings in the footer.
![image](https://github.com/SU-SWS/taiwan_hub_subtheme/assets/49299083/0e12a3f4-a2aa-4b41-874f-4585c26cd507)
- Hero adjustments per design QA
- Change Search results page input to a button in order to get correct button styles
- General improvements to the search results page at `/search` to give the links more space, see before and after
Before:
![Screenshot 2024-06-13 at 8 38 06 AM](https://github.com/SU-SWS/taiwan_hub_subtheme/assets/49299083/7448748d-9f52-48ab-94b6-9709941a2f3c)

After:
![Screenshot 2024-06-13 at 11 25 42 AM](https://github.com/SU-SWS/taiwan_hub_subtheme/assets/49299083/9c804cfe-7429-4b26-88f2-70fde407fd08)

- I made an A11y fixup for the lockup for the page at `/user`
![image](https://github.com/SU-SWS/taiwan_hub_subtheme/assets/49299083/fc7029b0-aab2-403d-9992-01ae7da594d7)


# Review By (Date)
- EOD 6/14

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Create a basic page and test out the headings in a Text Area Paragraph. Confirm the styles match Figma.
3. Navigate to `/search` and confirm the input in now a button with the correct styles
4. Confirm the top hero banner still looks good after these adjustments
5. Review code 🍞 

# Associated Issues and/or People
- [THUB-17](https://stanfordits.atlassian.net/browse/THUB-17): Text Area

[THUB-17]: https://stanfordits.atlassian.net/browse/THUB-17?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[THUB-17]: https://stanfordits.atlassian.net/browse/THUB-17?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

jenbreese commented:
I noticed a couple of things.
1. If I added a heading to the footer, I don't get the heading styles with the line, but the font size/type also changes. 
2. The heading sizes are a little smaller than the Figma.
<img width="1671" alt="Screenshot 2024-06-13 at 8 42 01 AM" src="https://github.com/SU-SWS/taiwan_hub_subtheme/assets/1187335/17c62930-229e-4633-bbd0-54220c98b440">
<img width="650" alt="Screenshot 2024-06-13 at 8 40 45 AM" src="https://github.com/SU-SWS/taiwan_hub_subtheme/assets/1187335/35af274d-c448-425b-9a89-d58dd41d7f40">

jdwjdwjdw commented:
> I noticed a couple of things.
> 
> 1. If I added a heading to the footer, I don't get the heading styles with the line, but the font size/type also changes.
> 2. The heading sizes are a little smaller than the Figma.

Thanks @jenbreese! I created type mixins to match Figma and applied them where relevant. The footer headings shouldn't have the gradient badge, but I fixed the font-sizes to match Figma better there.

I also included a couple of additional fixups:
- I made some improvements to the search results page at `/search` to give the links more space, see before and after

Before:
![Screenshot 2024-06-13 at 8 38 06 AM](https://github.com/SU-SWS/taiwan_hub_subtheme/assets/49299083/7448748d-9f52-48ab-94b6-9709941a2f3c)

After:
![Screenshot 2024-06-13 at 11 25 42 AM](https://github.com/SU-SWS/taiwan_hub_subtheme/assets/49299083/9c804cfe-7429-4b26-88f2-70fde407fd08)

- I made an A11y fixup for the lockup for the page at `/user`
![image](https://github.com/SU-SWS/taiwan_hub_subtheme/assets/49299083/fc7029b0-aab2-403d-9992-01ae7da594d7)

jenbreese reviewed (approved):
Thank you! Looks good to me.
